doc/coq.rst

```diff
@@ -0,0 +1,62 @@
+.. _coq-main:
+
+******
+Coq
+******
+
+Dune is also able to build Coq developments. A Coq project is a mix of
+Coq ``.v`` files and (optionally) OCaml libraries (in which case we say the project is a *Coq plugin*). To enable
```

Comment:
Actually, a Coq project could contain OCaml files without being a plugin. There is a specific (and standard case) which is an OCaml project where some OCaml files are automatically extracted from Coq files, and some others are hand-written. At some point, it would be good to have good Dune support for this.

A Coq plugin is a Coq project containing OCaml files that link to the Coq API.

Reply:
I will refine this a bit; but indeed I will not talk about extraction until we get support for it.